ejb3/src/main/java/org/jboss/as/ejb3/tx/CMTTxInterceptor.java

```diff
@@ -83,15 +83,16 @@ protected void endTransaction(final TransactionManager tm, final Transaction tx)
                 throw EjbLogger.ROOT_LOGGER.wrongTxOnThread(tx, tm.getTransaction());
             }
             final int txStatus = tx.getStatus();
-            if (txStatus == Status.STATUS_ACTIVE) {
+            if (txStatus == Status.STATUS_ACTIVE || txStatus == Status.STATUS_PREPARED || txStatus == Status.STATUS_COMMITTED) {
```

Comment:
I think you need preparing and committing in here too because those would indicate that someone is committing this in parallel so we need to sync with them